### DISCUSSION TITLE
Compatible with react hooks

### DISCUSSION THREAD

**Discussion by hec9527** [preactjs/preact#4315] [Q&A] [answered]
How to be compatible with react hooks? From the case given on the official website, after adding alias, I found that useEffect cannot be found. Do I need to install other packages to achieve compatibility?

![image](https://github.com/preactjs/preact/assets/28671232/4b8155eb-13ff-41d2-8fcc-5afdd718c8bf)

**Comment by marvinhagemeister** [comment]
Sounds like you didn't set up the react alias properly. What did you point it to?

**Reply by hec9527** [reply]
I did,  When I import hooks separately, there is no problem, for example: 
```js
import { useEffect, useRef, useState } from 'preact/hooks';
```

**Reply by marvinhagemeister** [reply]
No, we do export hooks from `preact/compat` as well. So if you have set up your react aliases correctly, then you can import hooks via:

```ts
import { useEffect, useRef, useState } from "react"
```

If this is not possible in your project, then you haven't set up the aliases properly, or something else is wrong. To help you further we need to rule out that aliases are set up correctly. Please share the value you pointed the `react` alias to.

**Reply by hec9527** [reply]
Thanks, I'll check the alias configuration first

**Reply by marvinhagemeister** [reply]
List of things to check:

1. Are aliases set up correctly? If no then the problem is there:
2. If they are set up correctly than we're missing a re-export as hooks should be exported. Looking at the output you posted earlier it looks like it doesn't alias to `preact/compat` directly but rather something else. For us to know where we would need to add those missing imports we would need to know how you set up your aliases and what you pointed them to.

Once both of these pieces information are available it should be pretty easy to resolve this.

**Reply by rschristian** [reply, marked as answer]
I'm 99% sure you're using `preact-compat`, instead of the correct `preact/compat`, in your aliases. `preact-compat` is legacy, for Preact v8 and older. `preact/compat` is the modern location of compat.

**Reply by hec9527** [reply]
Sorry, I made a stupid mistake and was misled by a wrong mirror website. Thank you for your correction

**Reply by rschristian** [reply]
No worries, it's easy to do! I've mistyped that a good handful of times even though I'm very familiar with them both.
